service/controller/release/resource/label/create.go

```diff
@@ -0,0 +1,70 @@
+package status
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/release-operator/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToReleaseCR(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding if custom resource needs to be updated")
+
+		currentLabels := cr.Labels
+		if currentLabels == nil {
+			currentLabels = map[string]string{}
+		}
+
+		v := cr.Labels[key.LabelReleaseCyclePhase]
```

Comment:
I think we should check if the label even exists. Above you just initialized the label map.

Reply:
Yeah this should be:

```suggestion
		v := currentLabels[key.LabelReleaseCyclePhase]
```

Zero value of the string is `""` so no problem with value not existing in the map.